builder/builder.go

```diff
@@ -156,3 +222,76 @@ func Receipt(fee *felt.Felt, feeUnit core.FeeUnit, txHash *felt.Felt, trace *vm.
 		RevertReason:       trace.RevertReason(),
 	}
 }
+
+func (b *Builder) listenPool(ctx context.Context) error {
+	for {
+		if err := b.depletePool(ctx); err != nil {
+			if !errors.Is(err, db.ErrKeyNotFound) {
+				return err
+			}
+		}
+
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-b.pool.Wait():
+			continue
+		}
+	}
+}
+
+func (b *Builder) depletePool(ctx context.Context) error {
+	for {
+		userTxn, err := b.pool.Pop()
```

Comment:
Does this return db.ErrKeyNotFound is the pool is empty? What do you think about opening a separate PR so that `Pop() `returns `txn,ok,err`?

Reply:
> What do you think about opening a separate PR so that Pop() returns txn,ok,err?

What do you think the benefit would be? Compared to checking if `err` is  `db.ErrKeyNotFound`